timemachine/fe/rbfe.py

```diff
@@ -579,8 +577,17 @@ def estimate_relative_free_energy_bisection(
 
         plots = make_pair_bar_plots(final_result, temperature, combined_prefix)
 
-        stored_frames = [np.array(frames[i]) for i in keep_idxs]
-        stored_boxes = [boxes[i] for i in keep_idxs]
+        assert len(results) == len(frames) == len(boxes)
+        stored_frames = []
+        stored_boxes = []
+        for i in keep_idxs:
+            try:
+                stored_frames.append(np.array(frames[i]))
```

Comment:
Does this convert the StoredFrames to np.array? Looks like we were doing this previously, but means we could run out of memory trying to get all the frames.

Reply:
Yes - so we definitely wouldn't want to set `keep_idxs=range(n_windows)`, at least for typical system sizes. I don't think the code here has ever handled this in a robust way. Should probably be a task for deboggling?

Reply:
Sounds good, just had thought we were returning StoredFrames in SimulationResults.